core/services/relay/evm/write_target.go

```diff
@@ -69,7 +83,192 @@ func NewWriteTarget(ctx context.Context, relayer *Relayer, chain legacyevm.Chain
+	if err != nil {
+		return "", err
+	}
+	var workflowName [10]byte
+	copy(workflowName[:], decodedName)
+	if !bytes.Equal(reportMetadata.WorkflowName[:], workflowName[:]) {
+		return "", fmt.Errorf("WorkflowName in the report does not match WorkflowName in the request metadata. Report WorkflowName: %+v, request WorkflowName: %+v", hex.EncodeToString(reportMetadata.WorkflowName[:]), hex.EncodeToString(workflowName[:]))
+	}
+
+	if hex.EncodeToString(reportMetadata.WorkflowCID[:]) != rawRequest.Metadata.WorkflowID {
+		return "", fmt.Errorf("WorkflowID in the report does not match WorkflowID in the request metadata. Report WorkflowID: %+v, request WorkflowID: %+v", reportMetadata.WorkflowCID, rawRequest.Metadata.WorkflowID)
+	}
+
+	if !bytes.Equal(reportMetadata.ReportID[:], r.Inputs.SignedReport.ID) {
+		return "", fmt.Errorf("ReportID in the report does not match ReportID in the inputs. reportMetadata.ReportID: %x, Inputs.SignedReport.ID: %x", reportMetadata.ReportID, r.Inputs.SignedReport.ID)
+	}
+
+	return r.Config.Address, nil
+}
+
+func decodeReportMetadata(data []byte) (metadata ReportV1Metadata, err error) {
+	if len(data) < metadata.Length() {
+		return metadata, fmt.Errorf("data too short: %d bytes", len(data))
+	}
+	return metadata, binary.Read(bytes.NewReader(data[:metadata.Length()]), binary.BigEndian, &metadata)
+}
+
+func (rm ReportV1Metadata) Encode() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	err := binary.Write(buf, binary.BigEndian, rm)
+	if err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+func (rm ReportV1Metadata) Length() int {
+	bytes, err := rm.Encode()
+	if err != nil {
+		return 0
+	}
+	return len(bytes)
+}
+
+// Note: This should be a shared type that the OCR3 package validates as well
+type ReportV1Metadata struct {
+	Version             uint8
+	WorkflowExecutionID [32]byte
+	Timestamp           uint32
+	DonID               uint32
+	DonConfigVersion    uint32
+	WorkflowCID         [32]byte
+	WorkflowName        [10]byte
+	WorkflowOwner       [20]byte
+	ReportID            [2]byte
+}
+
+func getChainInfo(chainID uint64) (monitor.ChainInfo, error) {
+	chainSelector := chainselectors.EvmChainIdToChainSelector()[chainID]
+	chainFamily, err := chainselectors.GetSelectorFamily(chainSelector)
+	if err != nil {
+		return monitor.ChainInfo{}, fmt.Errorf("failed to get chain family for selector %d: %w", chainSelector, err)
+	}
+	chainDetails, err := chainselectors.GetChainDetailsByChainIDAndFamily(strconv.Itoa(int(chainID)), chainFamily)
+	if err != nil {
+		return monitor.ChainInfo{}, fmt.Errorf("failed to get chain details for chain %d and family %s: %w", chainID, chainFamily, err)
+	}
+
+	return monitor.ChainInfo{
+		ChainFamilyName: chainFamily,
+		ChainID:         strconv.Itoa(int(chainID)),
+		NetworkName:     chainDetails.ChainName,
+		// TODO: not sure what the difference between NetworkName and NetworkNameFull is
+		NetworkNameFull: "",
```

Comment:
https://github.com/smartcontractkit/chainlink-aptos/blob/29eada1f85c67c9601cfe61d1c0ca35fa7c0d75e/scripts/core.config.toml#L28

- NetworkName: localnet or testnet or mainnet
- NetworkNameFull: aptos-localnet
    - Basically how we name our chain-selector: https://github.com/smartcontractkit/chain-selectors/blob/main/selectors_aptos.yml

Reply:
Ok, I don't think EVM has this concept afaik, I think keeping `NetworkNameFull` empty should be alright though

Reply:
But aren't these EVM selectors: https://github.com/smartcontractkit/chain-selectors/blob/main/selectors.yml

And your `chainDetails.ChainName` is basically the `NetworkNameFull`?

Reply:
ok so then we should just set `NetworkNameFull `to `chainDetails.ChainName` and leave `NetworkName` empty then? Or set them both to `chainDetails.ChainName`. This is just for beholder attributes so I don't think it is a critical change

Reply:
Not setting this correctly will make alerts not work: https://github.com/smartcontractkit/alerts/blob/1b69967a9b413716f7792b6d22605d4e0ea3a845/charts/prometheus-rules/rules/prometheus/data-feeds/data-feeds-aptos.yml#L28

Makes sense?

Reply:
I'm not sure if it's possible to distinguish between chain name and chain name full on EVM, do you know? Also, those are Aptos specific alerts so we could tweak them if needed to work with what we have on EVM

Reply:
No these are not Aptos specific alerts in any way